File: projet-java/brasil-burger-exam/src/main/java/com/brasilburger/entity/Menu.java
package com.brasilburger.entity;

import java.util.ArrayList;
import java.util.List;

public class Menu {
    private int id;
    private String nom;
    private String imageUrl;
    private String etat;
    private double prix;
    private List<MenuDetail> details = new ArrayList<>();

    public Menu() {}

    public Menu(String nom,double prix, String imageUrl,String etat) {
        this.nom = nom;
        this.imageUrl = imageUrl;
        this.etat = "DISPONIBLE";
        this.prix = prix;
        
    }


    public void addDetail(MenuDetail d) {
        details.add(d);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public List<MenuDetail> getDetails() {
        return details;
    }

    public double getPrix() {
        return prix;
    }
    public void setPrix(double prix) {
        this.prix = prix;
    }

    
}

File: src/Entity/Menu.php
<?php

namespace App\Entity;

use App\Repository\MenuRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Enum\EtatType;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Entity\CommandeItem;
use App\Entity\MenuDetail;



#[ORM\Entity(repositoryClass: MenuRepository::class)]
#[ORM\Table(name: "menu")]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $nom = null;

    #[ORM\Column(name: "image_url", type: "string", length: 500, nullable: true)]
    private ?string $imageUrl = null;

    #[ORM\Column(type: "decimal", precision: 10, scale: 2)]
    private string $prix;

    #[ORM\Column(enumType: \App\Enum\EtatType::class)]
    private EtatType $etat;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: CommandeItem::class)]
    private Collection $commandeItems;

    #[ORM\OneToMany(mappedBy: 'menu', targetEntity: MenuDetail::class)]
    private Collection $details;


    public function __construct()
    {
        $this->etat = EtatType::DISPONIBLE;
        $this->prix = '0';
        $this->commandeItems = new ArrayCollection();
        $this->details = new ArrayCollection();

    }


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getImageUrl(): ?string
    {
        return $this->imageUrl;
    }

    public function setImageUrl(string $imageUrl): static
    {
        $this->imageUrl = $imageUrl;

        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getEtat(): EtatType
    {
        return $this->etat;
    }

    public function setEtat(EtatType $etat): self
    {
        $this->etat = $etat;
        return $this;
    }
}
